scanspec/core.py

```diff
@@ -77,6 +77,28 @@ def alternative_constructor(f):
 generic_name = type_name(lambda cls, *args: cls.__name__)
 
 
+def to_gql_input(ob) -> str:
+    """Convert plain Python objects to their GraphQL representation.
+
+    >>> to_gql_input({"a": {"b": 1, "c": True, "d": 4.2, "e": "e"}})
+    '{a: {b: 1, c: true, d: 4.2, e: "e"}}'
+    """
+    if isinstance(ob, dict):
+        inner = ", ".join(f"{k}: {to_gql_input(v)}" for k, v in ob.items())
+        return "{%s}" % inner
+    elif isinstance(ob, list):
+        inner = ", ".join(to_gql_input(v) for v in ob)
+        return "[%s]" % inner
+    elif isinstance(ob, str):
+        return '"%s"' % ob
+    elif isinstance(ob, bool):
+        return "true" if ob else "false"
+    elif isinstance(ob, (int, float)):
+        return str(ob)
```

Comment:
Could this part be simplified?

```suggestion
    elif isinstance(ob, (str, int, float, bool)):
        return json.dumps(ob)
```